fideslog/api/routes/registrations.py

```diff
@@ -0,0 +1,151 @@
+from logging import getLogger
+from typing import List, Optional
+
+from fastapi import APIRouter, Depends, Query, Request, Response, status
+from sqlalchemy.exc import DBAPIError, NoResultFound
+from sqlalchemy.orm import Session
+from sqlalchemy.orm.exc import UnmappedInstanceError
+
+from ..database import get_db
+from ..database.registrations import create, delete, get, update
+from ..errors import InternalServerError, NotFoundError, TooManyRequestsError
+from ..schemas.registration import Registration
+
+log = getLogger(__name__)
+registration_router = APIRouter(tags=["Registrations"], prefix="/registrations")
+
+
+@registration_router.get(
+    "",
+    response_description="A list of registrations",
+    response_model=List[Registration],
+    responses={
+        status.HTTP_429_TOO_MANY_REQUESTS: TooManyRequestsError.doc(),
+        status.HTTP_500_INTERNAL_SERVER_ERROR: InternalServerError.doc(),
+    },
+    status_code=status.HTTP_200_OK,
+)
+async def list_registrations(
+    _: Request,
+    count: Optional[int] = Query(
+        default=None,
+        description="The amount of registrations to return.",
+        gt=0,
+    ),
+    offset: Optional[int] = Query(
+        default=None,
+        description="The amount of registrations to skip before returning.",
+        gt=0,
+    ),
+    database: Session = Depends(get_db),
+) -> List[Registration]:
```

Comment:
Appreciate this is a one-off, if we want to make this a more permanent feature in Fideslog we should consider that Fides uses [this lib](https://github.com/ethyca/fides/blob/main/src/fides/api/ops/api/v1/endpoints/policy_endpoints.py#L5-L7) to handle pagination